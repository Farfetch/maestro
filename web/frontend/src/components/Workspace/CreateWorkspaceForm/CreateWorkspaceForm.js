import { Form, Input, Select } from "antd";
import PropTypes from "prop-types";
import React from "react";

const CreateWorkspaceForm = ({ form, initialValues, onFinish, users }) => {
  const onFinishFailed = async () => {};

  return (
    <Form
      layout="vertical"
      form={form}
      name="basic"
      onFinish={onFinish}
      onFinishFailed={onFinishFailed}
      initialValues={initialValues}
    >
      <Form.Item label="Workspace Name">
        <Form.Item
          name="name"
          noStyle
          rules={[
            {
              required: true,
              message: "Please enter Name"
            }
          ]}
        >
          <Input placeholder="Enter prefered workspace name" />
        </Form.Item>
      </Form.Item>
      <Form.Item label="Users">
        <Form.Item
          name="usersEmail"
          noStyle
          rules={[
            {
              required: true,
              message: "Please select at leat one user"
            }
          ]}
        >
          <Select
            mode="tags"
            allowClear
            style={{ width: "100%" }}
            placeholder="Select or add new users to workspace"
          >
            {users.map((user) => (
              <Select.Option key={user.email}>{user.email}</Select.Option>
            ))}
          </Select>
        </Form.Item>
      </Form.Item>
    </Form>
  );
};

CreateWorkspaceForm.propTypes = {
  form: PropTypes.any.isRequired,
  initialValues: PropTypes.shape({
    name: PropTypes.string,
    usersEmail: PropTypes.arrayOf(PropTypes.string)
  }),
  onFinish: PropTypes.func.isRequired,
  users: PropTypes.arrayOf(
    PropTypes.shape({
      email: PropTypes.string.isRequired
    })
  )
};

export default CreateWorkspaceForm;
