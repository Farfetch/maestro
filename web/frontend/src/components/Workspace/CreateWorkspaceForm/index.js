import CreateWorkspaceForm from "./CreateWorkspaceForm";

export default CreateWorkspaceForm;
