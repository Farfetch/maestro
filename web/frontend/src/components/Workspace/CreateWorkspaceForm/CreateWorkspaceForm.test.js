import { cleanup, render, waitFor } from "@testing-library/react";
import React from "react";

import CreateWorkspaceForm from "./CreateWorkspaceForm";

afterEach(cleanup);

describe("components/Workspace/CreateWorkspaceForm", () => {
  test(`should render CreateWorkspaceForm component`, async () => {
    const users = [
      {
        id: "user-id",
        email: "user1@maestro.net"
      }
    ];

    const { container } = render(
      <CreateWorkspaceForm
        form={null}
        initialValues={{}}
        onFinish={() => null}
        users={users}
      />
    );

    await waitFor(() => {
      expect(container.outerHTML).toMatchSnapshot();
    });
  });

  test(`should render CreateWorkspaceForm component with initialValues`, async () => {
    const users = [
      {
        id: "user-id",
        email: "user1@maestro.net"
      },
      {
        id: "user-id-2",
        email: "user2@maestro.net"
      },
      {
        id: "user-id-3",
        email: "user3@maestro.net"
      }
    ];

    const { container } = render(
      <CreateWorkspaceForm
        form={null}
        initialValues={{
          name: "Some default value",
          usersEmail: ["user1@maestro.net", "user3@maestro.net"]
        }}
        onFinish={() => null}
        users={users}
      />
    );

    await waitFor(() => {
      expect(container.outerHTML).toMatchSnapshot();
    });
  });
});
