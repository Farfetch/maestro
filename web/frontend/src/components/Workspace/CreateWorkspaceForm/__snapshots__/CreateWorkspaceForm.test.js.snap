// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Workspace/CreateWorkspaceForm should render CreateWorkspaceForm component 1`] = `"<div><form id=\\"basic\\" class=\\"ant-form ant-form-vertical\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label class=\\"\\" title=\\"Workspace Name\\">Workspace Name</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><input placeholder=\\"Enter prefered workspace name\\" id=\\"basic_name\\" class=\\"ant-input\\" type=\\"text\\" value=\\"\\"></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label class=\\"\\" title=\\"Users\\">Users</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-select ant-select-multiple ant-select-allow-clear ant-select-show-search\\" style=\\"width: 100%;\\"><div class=\\"ant-select-selector\\"><div class=\\"ant-select-selection-overflow\\"><div class=\\"ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix\\" style=\\"opacity: 1;\\"><div class=\\"ant-select-selection-search\\" style=\\"width: 0px;\\"><input type=\\"search\\" id=\\"basic_usersEmail\\" autocomplete=\\"off\\" class=\\"ant-select-selection-search-input\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"basic_usersEmail_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"basic_usersEmail_list\\" aria-activedescendant=\\"basic_usersEmail_list_0\\" value=\\"\\"><span class=\\"ant-select-selection-search-mirror\\" aria-hidden=\\"true\\">&nbsp;</span></div></div></div><span class=\\"ant-select-selection-placeholder\\">Select or add new users to workspace</span></div></div></div></div></div></div></form></div>"`;

exports[`components/Workspace/CreateWorkspaceForm should render CreateWorkspaceForm component with initialValues 1`] = `"<div><form id=\\"basic\\" class=\\"ant-form ant-form-vertical\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label class=\\"\\" title=\\"Workspace Name\\">Workspace Name</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><input placeholder=\\"Enter prefered workspace name\\" id=\\"basic_name\\" class=\\"ant-input\\" type=\\"text\\" value=\\"Some default value\\"></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label class=\\"\\" title=\\"Users\\">Users</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-select ant-select-multiple ant-select-allow-clear ant-select-show-search\\" style=\\"width: 100%;\\"><div class=\\"ant-select-selector\\"><div class=\\"ant-select-selection-overflow\\"><div class=\\"ant-select-selection-overflow-item\\" style=\\"opacity: 1;\\"><span class=\\"ant-select-selection-item\\" title=\\"user1@maestro.net\\"><span class=\\"ant-select-selection-item-content\\">user1@maestro.net</span><span class=\\"ant-select-selection-item-remove\\" style=\\"user-select: none;\\" unselectable=\\"on\\" aria-hidden=\\"true\\"><span role=\\"img\\" aria-label=\\"close\\" class=\\"anticon anticon-close\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"close\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z\\"></path></svg></span></span></span></div><div class=\\"ant-select-selection-overflow-item\\" style=\\"opacity: 1;\\"><span class=\\"ant-select-selection-item\\" title=\\"user3@maestro.net\\"><span class=\\"ant-select-selection-item-content\\">user3@maestro.net</span><span class=\\"ant-select-selection-item-remove\\" style=\\"user-select: none;\\" unselectable=\\"on\\" aria-hidden=\\"true\\"><span role=\\"img\\" aria-label=\\"close\\" class=\\"anticon anticon-close\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"close\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z\\"></path></svg></span></span></span></div><div class=\\"ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix\\" style=\\"opacity: 1;\\"><div class=\\"ant-select-selection-search\\" style=\\"width: 0px;\\"><input type=\\"search\\" id=\\"basic_usersEmail\\" autocomplete=\\"off\\" class=\\"ant-select-selection-search-input\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"basic_usersEmail_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"basic_usersEmail_list\\" aria-activedescendant=\\"basic_usersEmail_list_0\\" value=\\"\\"><span class=\\"ant-select-selection-search-mirror\\" aria-hidden=\\"true\\">&nbsp;</span></div></div></div></div><span class=\\"ant-select-clear\\" style=\\"user-select: none;\\" unselectable=\\"on\\" aria-hidden=\\"true\\"><span role=\\"img\\" aria-label=\\"close-circle\\" class=\\"anticon anticon-close-circle\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"close-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z\\"></path></svg></span></span></div></div></div></div></div></form></div>"`;
